### PR TITLE
SMO-472 Basic UI for login screen + app UI architecture

### PR DESCRIPTION
**Pull request by OlehPiskorskyj**
Simulator Screenshot - iPhone 14 Pro - 2023-09-11 at 13 24 38
Simulator Screenshot - iPhone 14 Pro - 2023-09-11 at 13 24 50

### CODE REVIEW

File: SenacorOnlineBanking/ColorAssets.swift [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+//  ColorAssets.swift
+//  SenacorOnlineBanking
+//
+//  Created by Oleh Piskorskyi on 11/09/2023.
+//
+
+import Foundation
+
+enum ColorAssets: String {
+    case action
```
Comment (pawel-finanteq) [inline SenacorOnlineBanking/ColorAssets.swift:11]: Please add description how to use this colors

File: SenacorOnlineBanking/Coordinators/AppCoordinator.swift [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+//
+
+import SwiftUI
+import UIKit
+
+final class AppCoordinator: Coordinator {
+
+    var childCoordinators: [Coordinator] = []
+    var navigationStack = UINavigationController()
+    var window: UIWindow?
```
Comment (pawel-finanteq) [inline SenacorOnlineBanking/Coordinators/AppCoordinator.swift:15]: Please add `private`

File: SenacorOnlineBanking/Coordinators/AppCoordinator.swift [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+//  SenacorOnlineBanking
+//
+//  Created by Oleh Piskorskyi on 08/09/2023.
+//
+
+import SwiftUI
+import UIKit
+
+final class AppCoordinator: Coordinator {
+
```
Comment (pawel-finanteq) [inline SenacorOnlineBanking/Coordinators/AppCoordinator.swift:12]: In some places you are adding this leading empty line. I would recommend to remove all leading empty line in all places

File: SenacorOnlineBanking/Coordinators/AppCoordinator.swift (hunk 2 above)
Comment (pawel-finanteq) [reply SenacorOnlineBanking/Coordinators/AppCoordinator.swift:15]: Also, please remove optional

File: SenacorOnlineBanking/Screens/Login/LoginViewModel.swift [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+protocol LoginViewModelCoordinatorDelegate: AnyObject {
+    func showLoginConfirmationView(sender: LoginViewModel)
+}
+
+final class LoginViewModel: RendererViewModel, ObservableObject {
+    
+    @Published var email: String = ""
+    @Published var password: String = ""
+    weak var coordinatorDelegate: LoginViewModelCoordinatorDelegate?
```
Comment (pawel-finanteq) [inline SenacorOnlineBanking/Screens/Login/LoginViewModel.swift:19]: Please add `private`

File: SenacorOnlineBanking/Screens/LoginConfirmation/LoginConfirmationView.swift [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+//  SenacorOnlineBanking
+//
+//  Created by Oleh Piskorskyi on 11/09/2023.
+//
+
+import SwiftUI
+
+struct LoginConfirmationView: View {
+    
+    let viewModel: LoginConfirmationViewModel
```
Comment (pawel-finanteq) [inline SenacorOnlineBanking/Screens/LoginConfirmation/LoginConfirmationView.swift:12]: Please add `private

File: SenacorOnlineBanking/Screens/RendererViewModel.swift [hunk 6]
```diff
@@ -0,0 +1,10 @@
+//
+//  RendererViewModel.swift
+//  SenacorOnlineBanking
+//
+//  Created by Oleh Piskorskyi on 08/09/2023.
+//
+
+import Foundation
```
Comment (pawel-finanteq) [inline SenacorOnlineBanking/Screens/RendererViewModel.swift:8]: Please remove unnecessary import

File: SenacorOnlineBanking/Screens/LoginConfirmation/LoginConfirmationViewModel.swift [hunk 7]
```diff
@@ -0,0 +1,10 @@
+//
+//  LoginConfirmationViewModel.swift
+//  SenacorOnlineBanking
+//
+//  Created by Oleh Piskorskyi on 11/09/2023.
+//
+
+import Foundation
```
Comment (pawel-finanteq) [inline SenacorOnlineBanking/Screens/LoginConfirmation/LoginConfirmationViewModel.swift:8]: Please remove unnecessary import

File: SenacorOnlineBankingTests/Extensions/StringExtensionsTests.swift [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+//  Created by Oleh Piskorskyi on 11/09/2023.
+//
+
+import XCTest
+@testable import SenacorOnlineBanking
+
+final class StringExtensionsTests: XCTestCase {
+    
+    func test_isValidEmail() {
+        XCTAssertFalse("".isValidEmail)
```
Comment (pawel-finanteq) [inline SenacorOnlineBankingTests/Extensions/StringExtensionsTests.swift:14]: I would recommend to make one assert per test. This approach makes test results more readable

**Review by pawel-finanteq — CHANGES_REQUESTED** [review]

File: SenacorOnlineBankingTests/Screens/LoginViewModelTests.swift [hunk 9]
```diff
@@ -0,0 +1,95 @@
...
+//
+//  Created by Oleh Piskorskyi on 11/09/2023.
+//
+
+import XCTest
+@testable import SenacorOnlineBanking
+
+final class LoginViewModelTests: XCTestCase {
+    
+    var sut: LoginViewModel!
```
Comment (pawel-finanteq) [inline SenacorOnlineBankingTests/Screens/LoginViewModelTests.swift:13]: Please add `private`

File: SenacorOnlineBankingTests/Screens/LoginViewModelTests.swift [hunk 10]
```diff
@@ -0,0 +1,95 @@
...
+    var coordinatorDelegate: StubLoginViewModelCoordinatorDelegate!
```
Comment (pawel-finanteq) [inline SenacorOnlineBankingTests/Screens/LoginViewModelTests.swift:14]: Please add `private`

**Review by pawel-finanteq — CHANGES_REQUESTED** [review]

File: SenacorOnlineBanking/Screens/Login/LoginViewModel.swift (hunk 4 above)
Comment (OlehPiskorskyj) [reply SenacorOnlineBanking/Screens/Login/LoginViewModel.swift:19]: Decided internally  to keep it as it is.

File: SenacorOnlineBanking/ColorAssets.swift (hunk 1 above)
Comment (OlehPiskorskyj) [reply SenacorOnlineBanking/ColorAssets.swift:16]: Added in the latest commit.

File: SenacorOnlineBankingTests/Extensions/StringExtensionsTests.swift (hunk 8 above)
Comment (OlehPiskorskyj) [reply SenacorOnlineBankingTests/Extensions/StringExtensionsTests.swift:13]: Changed in the latest commit.

File: SenacorOnlineBankingTests/Screens/LoginViewModelTests.swift (hunk 9 above)
Comment (OlehPiskorskyj) [reply SenacorOnlineBankingTests/Screens/LoginViewModelTests.swift]: Changed in the latest commit.

File: SenacorOnlineBankingTests/Screens/LoginViewModelTests.swift (hunk 10 above)
Comment (OlehPiskorskyj) [reply SenacorOnlineBankingTests/Screens/LoginViewModelTests.swift]: Changed in the latest commit.

File: SenacorOnlineBanking/Coordinators/AppCoordinator.swift (hunk 3 above)
Comment (OlehPiskorskyj) [reply SenacorOnlineBanking/Coordinators/AppCoordinator.swift]: Fixed.

**Review by pawel-finanteq — APPROVED** [review]
This change can be made in next PR

**Pull request merged**